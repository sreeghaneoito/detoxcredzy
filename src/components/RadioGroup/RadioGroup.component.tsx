import {View, Text, FlatList, Pressable} from 'react-native';
import React, {useState} from 'react';
import {style} from './RadioGroup.style';
import {globalStyle} from '../../styles/global.style';
import {useUtils} from '../../hooks/useUtils';
export interface Props {
  withbg?: boolean;
  children?: any;
  Radiodata: {name: string; isSelected: boolean}[];
  testid?: string;
  setSelectedValue: (e: string) => void;
}
const RadioGroup = ({
  withbg,
  children,
  Radiodata,
  testid,
  setSelectedValue,
}: Props) => {
  const {theme} = useUtils();
  const [data, setData] = useState(Radiodata);
  const changeStatus = (indexDown: any) => {
    const SelectOption = data.map((obj, index) => {
      if (indexDown === index) {
        setSelectedValue(obj.name);
        return {...obj, isSelected: true};
      } else {
        return {...obj, isSelected: false};
      }
    });
    setData(SelectOption);
  };
  try {
    return (
      <View style={globalStyle.container} testID={testid}>
        <View>{children}</View>
        <FlatList
          style={style.radioGroup}
          data={data}
          renderItem={({item, index}) => {
            return (
              <Pressable
                style={[
                  style.radioBox,
                  withbg ? [style.withbg, theme.surface] : style.withoutbg,
                ]}
                onPress={() => {
                  changeStatus(index);
                }}
                testID={`radioitems${index}`}>
                <View style={style.radio}>
                  {item.isSelected && (
                    <View style={style.radioActive} testID="radioactive" />
                  )}
                </View>
                <Text style={theme.onBackground}>{item.name}</Text>
              </Pressable>
            );
          }}
        />
      </View>
    );
  } catch (error) {
    console.log('Rodio Component => ', error);
    return <View style={globalStyle.container} />;
  }
};

export default RadioGroup;
