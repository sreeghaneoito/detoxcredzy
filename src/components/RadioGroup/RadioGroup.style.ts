import {StyleSheet} from 'react-native';

export const style = StyleSheet.create({
  label: {
    fontSize: 16,
    color: 'black',
  },
  radioGroup: {
    paddingTop: 24,
  },
  radioBox: {
    marginVertical: 4,
    borderRadius: 4,
    alignItems: 'center',
    flexDirection: 'row',
  },
  radio: {
    width: 18,
    height: 18,
    borderRadius: 12,
    marginRight: 8,
    borderWidth: 1,
    borderColor: 'green',
    justifyContent: 'center',
    alignItems: 'center',
  },
  radioActive: {
    backgroundColor: 'green',
    width: 12,
    height: 12,
    borderRadius: 8,
  },
  withbg: {
    height: 48,
    paddingHorizontal: 16,
  },
  withoutbg: {
    height: 30,
  },
});
