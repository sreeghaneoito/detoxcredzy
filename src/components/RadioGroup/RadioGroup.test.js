import React from 'react';
import renderer from 'react-test-renderer';
import RadioGroup from './RadioGroup.component';

test('renders correctly', () => {
  const tree = renderer.create(<RadioGroup />).toJSON();
  expect(tree).toMatchSnapshot();
});
