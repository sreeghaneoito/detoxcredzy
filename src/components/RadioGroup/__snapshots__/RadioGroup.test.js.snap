// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginVertical": 20,
    }
  }
>
  <View />
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "paddingTop": 24,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
